src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -232,7 +232,7 @@ public static unsafe void Fill<T>(ref T refData, nuint numElements, T value)
 
             nint index = 0; // Use nint for arithmetic to avoid unnecessary 64->32->64 truncations
 
-            if (default(T) != null || (object?)value != null)
```

Comment:
Do these changes really have a material impact on the codegen?  I wouldn't expect them to.  The before/after in the opening post isn't for this case.